### DISCUSSION TITLE
How to cache the 'GET /me' route?

### DISCUSSION THREAD

**Discussion by saeedesmaili** [fastapi-users/fastapi-users#1315] [Q&A]
I'm using [fastapi-cache](https://github.com/long2ice/fastapi-cache) to cache certain endpoints of the app. I couldn't find a way to cache the `GET /me` route that fastapi-users provides. Is this possible?

Here is how I'm using fastapi-cache for other endpoints:

```python
from fastapi_cache.decorator import cache


@app.get("/hello")
@cache(expire=60)
async def hello_world():
    return "hello world"
```

**Comment by hgalytoby** [comment]
You can find the variable "current_active_user" in the documentation at this URL: https://fastapi-users.github.io/fastapi-users/12.1/configuration/full-example/#sqlalchemy.


```python
from fastapi_users import schemas


@app.get("/cache-me1")
@cache(expire=60)
async def cache_me(user: User = Depends(current_active_user)):
    print(user)
    return schemas.model_validate(UserRead, user)

```

You can also make partial modifications by referring to the source code at https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/router/users.py#L48.


```py
from fastapi_users import schemas


@app.get(
    "/cache-me2",
    response_model=UserRead,
    name="users:current_user",
    responses={
        status.HTTP_401_UNAUTHORIZED: {
            "description": "Missing token or inactive user.",
        },
    },
)
@cache(expire=60)
async def me(
    user: User = Depends(current_active_user),
):
    print(user)
    return schemas.model_validate(UserRead, user)
```


Here are all the routers: https://github.com/fastapi-users/fastapi-users/tree/master/fastapi_users/router. If you need customization, you can take a look.
